cloud/amazon/ec2_eni.py

```diff
@@ -63,6 +65,13 @@
       - The index of the device for the network interface attachment on the instance.
     required: false
     default: 0
+  attached:
+    description:
+      - Specifies if network interface should be attached or detached from instance. If attached=yes and no \
+        instance_id is given, attachment status won't change
```

Comment:
I don't think it's necessary to specify the attached=yes stuff here.  Just add another required_if statement in the argument spec